README.md

```diff
@@ -0,0 +1,843 @@
+---
+title: Testing workflow for web-components
+published: false
+description: Testing and debugging your web component is an important skill to give you the confidence the code you share is production ready.
+tags: webcomponents, javascript, testing, karma
+---
+
+Whenever you ship something to be used by others you take on a responsibility.
+Unfortunately not always is this responsibility taken care of.
+One thing to make sure you take it on is to write tests.
+
+No matter how small - no matter how simple there should be tests.
+*Yes I know reality hit's hard and there will be many cases where that doesn't happen - but strife for it*
+
+### Disclaimer
+We are going to make a simple version of an accessible input.
+Everything you see here is for illustration purposes only - so don't scream at me that you should not do it.
+We do it so we can see all the places where our testing workflow can shine.
+
+If you wanna play along - all the code is on [github](https://github.com/daKmoR/testing-workflow-for-web-components).
+
+## Let's get started
+
+After following [https://open-wc.org/testing/](https://open-wc.org/testing/) you should have the basic setup up and running.
+
+Let's create `src/a11y-input.js`;
+```js
+import { LitElement, html, css } from 'lit-element';
+
+export class A11yInput extends LitElement {}
+
+customElements.define('a11y-input', A11yInput);
+```
+
+and `test/a11y-input.test.js`;
+```js
+/* eslint-disable no-unused-expressions */
+import { html, fixture, expect } from '@open-wc/testing';
+
+import '../src/a11y-input.js';
+
+/**
+ * @typedef {import('../src/a11y-input.js').A11yInput} A11yInput
+ */
+
+describe('a11y input', () => {
+  it('has by default an empty string as label', async () => {
+    const el = /** @type {A11yInput} */ (await fixture('<a11y-input></a11y-input>'));
```

Comment:
Type definitions might confuse people. Perhaps its better to focus on just tests?